### PR TITLE
Support SteppingThreadSleeper

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Implements light thread sleeper to support invoke a sleeping sleeper to determine whether need continue to sleep. 

### Why are the changes needed?

Without this feature, when we need to refresh the interval of Heartbeat thread, it cannot take effect.

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Comment by maobaolong** [comment]
@jiacheliu3 @dbw9580 Would please take a look at this PR? Thank you!

**Comment by Jackson-Wang-7** [comment]
@jiacheliu3 @dbw9580 please take a look at this PR

**Comment by maobaolong** [comment]
Add a test to verify the sleeper can be updated.

**Comment by maobaolong** [comment]
@jiacheliu3 @dbw9580 Thank for the previous review, PTAL.

File: core/common/src/main/java/alluxio/time/SteppingThreadSleeper.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    if (duration.toMillis() < mSleepStepMs) {
+      sleep(duration);
+      return;
+    }
+    long startSleepMs = mClock.millis();
+    long sleepTo = startSleepMs + duration.toMillis();
+    long timeNow;
+    while ((timeNow = mClock.millis()) < sleepTo) {
+      mInternalSleeper.sleep(Duration.ofMillis(sleepTo - timeNow > mSleepStepMs
+          ? mSleepStepMs : sleepTo - timeNow));
```
Comment (dbw9580) [inline core/common/src/main/java/alluxio/time/SteppingThreadSleeper.java:70]:
```suggestion
      long sleepTime = Math.min(sleepTo - timeNow, mSleepStepMs);
      mInternalSleeper.sleep(Duration.ofMillis(sleepTime));
```

File: core/common/src/test/java/alluxio/heartbeat/SleepingTimerTest.java [hunk 2]
```diff
@@ -77,12 +85,45 @@ public void sleepForSpecifiedInterval() throws Exception {
...
 
     stimer.tick();
     mFakeClock.addTimeMs(INTERVAL_MS / 3);
     stimer.tick();
     verify(mMockSleeper).sleep(Duration.ofMillis(INTERVAL_MS - (INTERVAL_MS / 3)));
   }
+
+  @Test
+  public void updateIntervalForSteppingTimer() throws Exception {
+    AtomicLong interval = new AtomicLong(10000L);
```
Comment (dbw9580) [inline core/common/src/test/java/alluxio/heartbeat/SleepingTimerTest.java:100]:
```suggestion
    AtomicLong interval = new AtomicLong(10 * Constants.SECOND_MS);
```

**Review by dbw9580 — APPROVED** [review]
LGTM with a few minor comments

**Comment by maobaolong** [comment]
@jiacheliu3 Would you like to take another look?

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**